quinn/src/connection.rs

```diff
@@ -1255,7 +1255,7 @@ pub enum SendDatagramError {
 const MAX_TRANSMIT_DATAGRAMS: usize = 20;
 
 /// Error indicating that a stream has already been finished or reset
-#[derive(Debug, Error, Clone, PartialEq, Eq)]
+#[derive(Debug, Error, Clone, Copy, PartialEq, Eq)]
```

Comment:
Do you remember why this is a separate type?

Reply:
If you mean to ask why it's separate from the `proto` definition, perhaps that was motivated by trait impls that would otherwise have coherency issues? I'm not sure. If we can reuse the proto-layer type here I don't immediately see a drawback to doing so.

Reply:
> If you mean to ask why it's separate from the `proto` definition, perhaps that was motivated by trait impls that would otherwise have coherency issues? I'm not sure. If we can reuse the proto-layer type here I don't immediately see a drawback to doing so.

I wonder if we can do that semver-compatibly...

Reply:
I'd file it under "technically breaking but probably fine." Downstream could implement a downstream-defined trait separately for `proto::ClosedStream` and `quinn::ClosedStream`, which would break, but:
- why would you ever do that?
- it's trivial to fix by deleting either impl

Reply:
Okay, please have a look.